repolock.py

```diff
@@ -0,0 +1,122 @@
+# repolock.py - Endless OSTree repository locking
+#
+# Copyright (C) 2017  Endless Mobile, Inc <maintainers@endlessm.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import contextlib
+import errno
+import fcntl
+from itertools import imap
+from io import open
```

Comment:
Ah, OK. It appears that if you're on python 2.7 (I think we can make that assumption), then the difference is if you want to open in text mode and specify the encoding - https://stackoverflow.com/a/4897088. I don't think we're trying to open in text mode and actually read or write anything here, but it doesn't hurt to use `io.open`.